awsenv/replacer_test.go

```diff
@@ -133,3 +184,87 @@ type mockParamsGetter func(context.Context, []string) (map[string]string, error)
 func (f mockParamsGetter) GetParams(ctx context.Context, paths []string) (map[string]string, error) {
 	return f(ctx, paths)
 }
+
+func TestFilterPaths(t *testing.T) {
+	tests := []struct {
+		prefix string
+		input  map[string]string
+		want   []string
+	}{
+		{
+			prefix: "",
+			input:  nil,
+			want:   []string{},
+		},
+		{
+			prefix: "awsenv:",
+			input:  map[string]string{"X": "1", "Y": "pre:/y", "Z": "awsenv:/z"},
+			want:   []string{"/z"},
+		},
+		{
+			prefix: "pre:",
+			input:  map[string]string{"X": "1", "Y": "pre:/y", "Z": "awsenv:/z"},
+			want:   []string{"/y"},
+		},
+	}
+
+	for _, test := range tests {
+		r := Replacer{prefix: test.prefix}
+		got, want := r.filterPaths(test.input), test.want
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("filterPaths(%q) = %v, want %v", test.input, got, want)
+		}
+	}
+}
+
+func TestApplyPaths(t *testing.T) {
+	tests := []struct {
+		prefix            string
+		src               map[string]string
+		replaceWithValues map[string]string
+		want              map[string]string
+	}{
+		{
+			prefix:            "awsenv:",
+			src:               nil,
+			replaceWithValues: nil,
+			want:              nil,
+		},
+		{
+			prefix:            "awsenv:",
+			src:               map[string]string{},
+			replaceWithValues: map[string]string{"x": "a"},
+			want:              map[string]string{},
+		},
+		{
+			prefix:            "pre:",
+			src:               map[string]string{"x": "pre:/a", "y": "pre:/b", "z": "awsenv:/c"},
+			replaceWithValues: map[string]string{"/a": "A", "/b": "B", "/c": "C"},
+			want:              map[string]string{"x": "A", "y": "B", "z": "awsenv:/c"},
+		},
+		{
+			prefix:            "awsenv:",
+			src:               map[string]string{"x": "1"},
+			replaceWithValues: nil,
+			want:              map[string]string{"x": "1"},
+		},
+		{
+			prefix:            "awsenv:",
+			src:               map[string]string{"x": "awsenv:/a", "y": "awsenv:/b"},
+			replaceWithValues: map[string]string{"/a": "A", "/b": "B"},
+			want:              map[string]string{"x": "A", "y": "B"},
+		},
+	}
+
+	for idx, test := range tests {
+		test := test
+		t.Run(fmt.Sprintf("test_%v", idx), func(t *testing.T) {
```

Comment:
Should we add a `name` to the test struct so the failure message does not just include an index?